web/packages/teleterm/src/ui/Search/SearchContext.tsx

```diff
@@ -112,6 +122,12 @@ export const SearchContextProvider: FC = props => {
     try {
       await action;
     } finally {
+      // By the time the action passes, the user might have caused the focus to be lost on the
+      // search input, so let's bring it back.
+      //
+      // focus needs to be executed before the state update, otherwise the search bar will close for
+      // some reason.
```

Comment:
I couldn't find a way to write a test for this part. I think it has to do with how the browser and React interact together so it might be hard to reproduce it in a non-browser environment like our tests.

Reply:
Yeah, it can be possible.